xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -341,4 +321,28 @@ - (NSEvent*)InputEventHandler:(NSEvent*)nsevent
     return nullptr;
 }
 
+- (XBMC_Event)keyPressEvent:(CGEventRef*)event
+{
+  UniChar unicodeString[MAXUNICODESTRLEN] = {0};
+  UniCharCount actualStringLength = 0;
+
+  CGKeyCode keycode = (CGKeyCode)CGEventGetIntegerValueField(*event, kCGKeyboardEventKeycode);
+  CGEventKeyboardGetUnicodeString(*event, MAXUNICODESTRLEN,
+                                  &actualStringLength, unicodeString);
+
+  XBMC_Event newEvent = {};
+
+  if (actualStringLength == 0)
```

Comment:
it looks to me that you extracted the original code incorrectly. The following parts are missing:

```objc
// 1. when actualStringLength is 0
unicodeString[0] = [self OsxKey2XbmcKey:[nsevent.characters characterAtIndex:0]];

// 2
if (actualStringLength > 1)
  newEvent.key.keysym.unicode |= (unicodeString[1] << 8);
```

Reply:
Honestly I played with it and couldn't get a string longer than a single character. I've seen other snippets who just stick to a single char, really not sure which way to go

Reply:
ok, let's stick to a single character then. Maybe add a short comment in case we'll have to adjust it in the future.